flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/JobGraph.java

```diff
@@ -100,7 +101,7 @@
 	private final List<PermanentBlobKey> userJarBlobKeys = new ArrayList<>();
 
 	/** List of classpaths required to run this job. */
-	private List<URL> classpaths = Collections.emptyList();
```

Comment:
It's problematic that `JobGraph` implements `Serialized` and `classpaths` is a non-transient field. After a cluster upgrade to 1.10 `addClasspaths` will throw `UnsupportedOperationException` if `classpath` of the serialized JobGraph was previously empty.

Reply:
OK I can see the problem. Although we can work around by adding another field and add some bridge code I don't it is worth to do so because this is just a nice to have improvement. A reasonable way in my mind is that we take this thing into consideration when we port JobGraph to its protobuf version if there will be one. Anyway, defer the effort until we are in a codebase easy to do it.